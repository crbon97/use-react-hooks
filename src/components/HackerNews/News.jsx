import React from "react";

const News = ({ url, title }) => {
  return (
    <li>
      <a href={url} target={"_blank"}>
        {title}
      </a>
    </li>
  );
};

export default News;
