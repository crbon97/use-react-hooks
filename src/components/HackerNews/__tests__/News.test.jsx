import News from "../News";
import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

const testProps = {
  title: "React 16",
  url: "https://facebook.github.io/react/blog/2017/09/26/react-v16.0.html"
};
describe("Test News component with `react-test-renderer`", () => {
  test("News should render without error", () => {
    const newsWrapper = renderer.create(<News {...testProps} />);
    let tree = newsWrapper.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Test News component with `enzyme`", () => {
  test("News should render with the right props", () => {
    const newsWrapper = shallow(<News {...testProps} />);
    const linkElement = newsWrapper.find("a");
    expect(linkElement).toBeTruthy();
    expect(linkElement).toHaveLength(1);

    expect(linkElement.prop("href")).toEqual(testProps.url);
    expect(linkElement.text()).toEqual(testProps.title);
  });
});
