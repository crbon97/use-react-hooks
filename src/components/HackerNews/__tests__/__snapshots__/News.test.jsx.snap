// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test News component with \`react-test-renderer\` News should render without error 1`] = `
<li>
  <a
    href="https://facebook.github.io/react/blog/2017/09/26/react-v16.0.html"
    target="_blank"
  >
    React 16
  </a>
</li>
`;
